docs/rfc/rfc-026-p2p-bad-peers-checktx.md

```diff
@@ -0,0 +1,375 @@
+
+## Changelog
+- Nov 4, 2022: Initial draft (jmalicevic)
+- Nov 8, 2022: Updated draft (jmalicevic)
+
+## Abstract
+
+Before adding a transaction to its mempool, a node runs `CheckTx` on it. `CheckTx` returns a non-zero code
+in case the transaction fails an application specific check. 
+
+Note that there are valid scenarios in which a transaction will not
+pass this check (including nodes getting different transactions at different times, meaning some
+of them might be obsolete at the time of the check; state changes upon block execution etc.). 
+However, Tendermint users observed that
+there are transactions that can never have been or will never be valid. They thus propose
+to introduce a special response code for `CheckTx` to indicate this behaviour, and ban the peers
+who gossip such transactions. Additionally, users expressed a need for banning peers who
+repeadetly send transactions failing `CheckTx`. 
+
+The main goal of this document is to analyse the cases where peers could be banned when they send 
+transacations failing `CheckTx`, and provide the exact conditions that a peer and transaction have
+to satisfy in order to mark the peer as bad.
+
+This document will also include a proposal for implementing these changes within the mempool, including
+potential changes to the existing mempool logic and implementation.
+
+
+## Background
+
+This work was triggered by issue [#7918](https://github.com/tendermint/tendermint/issues/7918) and a related
+discussion in [#2185](https://github.com/tendermint/tendermint/issues/2185). Additionally,
+there was a [proposal](https://github.com/tendermint/tendermint/issues/6523) 
+to disconnect from peers after they send us transactions that constantly fail `CheckTx`.  While 
+the actual implementation of an additional response code for `CheckTx` is straight forward there 
+are certain things to consider. The questions to answer, along with identified risks will be outlined in 
+the discussion. 
+
+### Existing issues and concerns
+
+Before diving into the details, we collected a set of issues opened by various users, arguing for 
+this behaviour and explaining their needs. 
+
+- Celestia: [blacklisting peers that repeatedly send bad tx](https://github.com/celestiaorg/celestia-core/issues/867)
+  and investigating how [Tendermint treats large Txs](https://github.com/celestiaorg/celestia-core/issues/243)
+- BigChainDb: [Handling proposers who propose bad blocks](https://github.com/bigchaindb/BEPs/issues/84)
+- IBC relayer: Simulate transactions in the mempool and detect whether they could ever have been valid to 
+  [avoid overloading the p2p/mempool system](https://github.com/cosmos/ibc-go/issues/853#issuecomment-1032211020)
+ (ToDo check with Adi what is this about, what are valid reasons this can happen and when should we disconnect)
+  (relevant code [here](https://github.com/cosmos/ibc-go/blob/a0e59b8e7a2e1305b7b168962e20516ca8c98fad/modules/core/ante/ante.go#L23)) 
```

Comment:
Please double-check the text in this bullet:
* You say "IBC relayer" at the beginning, but the links are to the `ibc-go` component (which is combined with the SDK, it's not a relayer AFAIK)